dask_conversions/done/Dask_Land_Change_geomedian.ipynb

```diff
@@ -4,7 +4,39 @@
    "cell_type": "code",
```

Comment:
<p>Could you explain to me what the 'time' part is doing here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SatelliteApplicationsCatapult/odc-hub/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
<p>If we don't specify a size for time then we get a size of 1. So (time: 1, x: 1000, y: 1000) normally. If we have loads time data it means we process things slowly because we're copying loads of tiny chunks all over the place - it adds overhead.</p><p> </p><p>Increasing the size of the time chunk doesn't always speed us up. In this case it was the only thing which sped us up!</p><p> </p>

Reply:
You  can see it in action if you look at `xx_clean_b`. We have `chunksize=(9, 269, 210)`.

Reply:
<p>Cool - this makes sense - however, the input requirements for these notebook will be constantly be changing. E.g. someone may be interested in only 1 month the next time they may be interested in 2 years worth of data. The same goes for areas also - sometimes will only be interested in a small area, sometimes much larger.</p><p> </p><p>I do wonder if there is a way that we can determine the 'time' and the chunk size based on the users request, rather than hardcoding. E.g. could say if more then 30 scenes have been returned use a value of  5 for 'time', if only 10 images use a value of 1 for 'time', if 200 scenes use 10. I'm not sure how exactly that would work. Maybe a discussion for next week. </p>

Reply:
I think we don't need to worry too much about it. In general, larger chunks are better (the chunk sizes we're using here are already pretty small and making them smaller increases scheduler overheads) and chunk size is determined by the RAM on the cluster. So there's not much we'd want to change dynamically.

One situation worth considering is what happens when we have a data cube slightly larger than our max chunk size. Say we have a max of 1000x1000 but a data cube of 1200x1200. We'll have one large chunk and 3 tiny chunks. Then it could make sense to do something like 600x600 or 1200x600 so that we have equal size chunks.